src/packet_distribution.cpp

```diff
@@ -698,10 +698,24 @@ void AOApplication::server_packet_received(AOPacket *p_packet)
   else if (header == "AUTH") {
     if (!courtroom_constructed || !auth_packet_enabled || f_contents.size() < 1)
       goto end;
-    int authenticated = f_contents.at(0).toInt();
+    bool ok;
+    int authenticated = f_contents.at(0).toInt(&ok);
+    if (!ok)
+      goto end;
 
     w_courtroom->on_authentication_state_received(authenticated);
   }
+  else if (header == "JD") {
+    if (!courtroom_constructed || f_contents.size() < 1)
+      goto end;
+    bool ok;
+    int judnum = f_contents.at(0).toInt(&ok);
+    if (!ok)
+      goto end;
+    w_courtroom->set_jud(judnum);
+    if (w_courtroom->get_jud() > -1) // If we receive JD -1, it means the server asks us to fall back to client-side judge buttons behavior
```

Comment:
Yeah but... the clientside behavior isn't triggered?

Reply:
this is annoying and it's not what i intended when i wrote this. @Crystalwarrior you can deal with this

Reply:
upon closer inspection the clientside behavior is in fact triggered by line 717

Reply:
if `JD = -1`, nothing will happen... is that really what you want to happen?

Reply:
oh wait i'm dumb. no that's not what we want